game/views.py

```
@@ -103,6 +105,49 @@
+# These two functions are the core of the fix for Issue #2186.
+# Instead of reading/writing session keys like request.session['board'],
+# ALL game state is now stored under a game_id-scoped key:
+#   request.session[f'game_{game_id}'] = { 'board': ..., 'mode': ..., ... }
+# This means two tabs can each have their own game without interfering.
+# ──────────────────────────────────────────────────────────────────────────────
+
+# ─────────────────────────────────────────────────────────────────
+# SESSION HELPERS — Added for Issue #2186 (game_id scoping fix)
+# ─────────────────────────────────────────────────────────────────
+
+def _get_game_state(request, game_id):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_get_game_state`

(ANN202)